src/common/contracts/ao-contract.ts

```diff
@@ -0,0 +1,77 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc. All Rights Reserved.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+import { dryrun } from '@permaweb/aoconnect';
+
+import { AOContract, BaseContract, Logger } from '../../types.js';
+import { DefaultLogger } from '../logger.js';
+
+export class AOProcess<T> implements BaseContract<T>, AOContract {
+  private logger: Logger;
+  private processId: string;
+
+  constructor({
+    processId,
+    logger = new DefaultLogger(),
+  }: {
+    processId: string;
+    logger?: DefaultLogger;
+  }) {
+    this.processId = processId;
+    this.logger = logger;
+  }
+
+  async getState(): Promise<T> {
```

Comment:
Just a note on this, I think we are currently limited to 10mb returns here.

Reply:
[05b07cf](https://github.com/ar-io/ar-io-sdk/pull/108/commits/05b07cfbb1d974e708108c8239d8214d6c50b418)